libraries/botbuilder/tests/teams/adaptiveCards/src/adaptiveCardsBot.ts

```diff
@@ -0,0 +1,234 @@
+            if (text && text.length > 0) {
+                text = text.trim();
+                if (text == '1')
+                {
+                    await this.sendAdaptiveCard1(context);
+                }
+                else if (text == '2')
+                {
+                    await this.sendAdaptiveCard2(context);
+                }
+                else if (text == '3')
+                {
+                    await this.sendAdaptiveCard3(context);
+                }
+                else
+                {
+                    await context.sendActivity(`You said: ${text}`);
+                }
+            }
+            else {
+                await context.sendActivity('App sent a message with empty text');
+                const activityValue = context.activity.value;
+                if (activityValue) {
+                    await context.sendActivity(`but with value ${JSON.stringify(activityValue)}`);
+                }
+            }
+            await next();
+        });
+
+        this.onMembersAdded(async (context, next) => {
+            const membersAdded = context.activity.membersAdded;
+            for (const member of membersAdded) {
+                if (member.id !== context.activity.recipient.id) {
+                    await context.sendActivity('Hello and welcome!');
+                }
+            }
+
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+    }
+
+    protected async onTeamsTaskModuleFetch(context: TurnContext, taskModuleRequest: TaskModuleRequest): Promise<TaskModuleTaskInfo> {
+        await context.sendActivity(MessageFactory.text(`OnTeamsTaskModuleFetchAsync TaskModuleRequest: ${JSON.stringify(taskModuleRequest)}`));
+
+        const card = CardFactory.adaptiveCard({
+            "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+            "type": "AdaptiveCard",
+            "version": "1.0",
+            "body": [
+                {
+                   "type": "TextBlock",
+                   "text": "This is an Adaptive Card within a Task Module"
+                }
+            ],
+            "actions": [
+                {
+                    "type": "Action.Submit",
+                    "title": "Action.Submit",
+                    "data": {
+                        "submitLocation": "taskModule"
+                    }
+                }
+            ]
+        });
+
+        return <TaskModuleTaskInfo> {
+            card: CardFactory.adaptiveCard(card),
+            height: 200,
+            width: 400,
+            title: 'Task Module Example'
+        };
+    }
+
+    protected async onTeamsTaskModuleSubmit(context: TurnContext, taskModuleRequest: TaskModuleRequest): Promise<TaskModuleResponseBase> {
+        await context.sendActivity(MessageFactory.text(`OnTeamsTaskModuleSubmit value: ${ JSON.stringify(taskModuleRequest) }`));
+        return <TaskModuleMessageResponse>{ value: 'Thanks!' };
```

Comment:
Since this is just an interface in javascript, the 'type' of 'message' is required:

```js
return <TaskModuleMessageResponse>{ type: 'message', value: 'Thanks!' };
```